src/request/sign-swap/SignSwapApi.js

```diff
@@ -422,6 +438,36 @@ class SignSwapApi extends PolygonRequestParserMixin(BitcoinRequestParserMixin(To
         }
     }
 
+    /**
+     * Checks that the given instruction is a valid OASIS SettlementInstruction
+     * @param {unknown} obj
+     * @param {string} parameterName
+     * @returns {Omit<KeyguardRequest.SinpeMovilSettlementInstruction, 'contractId'>}
+     */
+    parseOasisCrcSettlementInstruction(obj, parameterName) {
+        if (typeof obj !== 'object' || obj === null) {
+            throw new Errors.InvalidRequestError('Invalid settlement');
+        }
+
+        const recipient = /** @type {{recipient: unknown}} */ (obj).recipient;
+        if (typeof recipient !== 'object' || recipient === null) {
+            throw new Errors.InvalidRequestError('Invalid settlement recipient');
+        }
+
+        /** @type {Omit<KeyguardRequest.SinpeMovilSettlementInstruction, 'contractId'>} */
+        const settlement = {
+            type: 'sinpemovil',
+            phoneNumber: /** @type {string} */ (
+                this.parseLabel(
+                    /** @type {{phoneNumber: unknown}} */ (recipient).phoneNumber,
+                    false,
+                    `${parameterName}.phoneNumber`,
+                )
+            ),
+        };
```

Comment:
This must now support `mock` type settlement, too, like `parseOasisSettlementInstruction` above.

Reply:
In fact, you should be able to reuse `parseOasisSettlementInstruction` with an additional switch case for `sinpemovil`.

Reply:
Also note that the `sinpemovil` settlement instruction doesn't have a `recipient` property, but has the `phoneNumber` directly in the root.

Reply:
I had some issues with types and generics.

I had to divide the switch to multiple functions. Not super elegant, but I don't want to overcomplicate generics types using just JSDoc. 

https://github.com/nimiq/keyguard/blob/onmax/sinpe/src/request/sign-swap/SignSwapApi.js#L414-L500

Reply:
I re-unified the functions into one.